Sources/RTMP/RTMPNWSocket.swift

```diff
@@ -103,16 +110,7 @@ final class RTMPNWSocket: RTMPSocketCompatible {
             events.append(Event(type: .rtmpStatus, bubbles: false, data: data))
         }
         readyState = .closing
-        if connection.state == .ready {
-            outputQueue.async {
-                let completion: NWConnection.SendCompletion = .contentProcessed { (_: Error?) in
-                    self.connection = nil
-                }
-                connection.send(content: nil, contentContext: .finalMessage, isComplete: true, completion: completion)
-            }
-        } else {
-            self.connection = nil
-        }
```

Comment:
This process was created specifically to address this issue, so simply removing it would cause the problem to reoccur.

Reply:
I see... What was happening is: connection was stale so end message could not be sent..  so it was stuck forever,  completion handler was never reached.

Maybe with viabilityUpdateHandler check for viability this could work